pennylane/gradients/parameter_shift.py

```diff
@@ -558,115 +567,124 @@ def _process_pda2_involutory(tape, pdA2, var_idx, non_involutory):
     """
     new_pdA2 = []
     for i in range(len(tape.measurements)):
-        if i in var_idx:
-            obs_involutory = i not in non_involutory
-            if obs_involutory:
-                num_params = len(tape.trainable_params)
+
+        if i in involutory_indices:
+            num_params = len(tape.trainable_params)
+            if num_params > 1:
                 item = tuple(np.array(0) for _ in range(num_params))
             else:
-                item = pdA2[i]
-            new_pdA2.append(item)
+                item = np.array(0)
+        else:
+            item = pdA2[i]
+        new_pdA2.append(item)
 
     return new_pdA2
 
 
-def _get_pdA2(results, tape, pdA2_fn, tape_boundary, non_involutory, var_idx):
+def _get_pdA2(results, tape, pdA2_fn, tape_boundary, non_involutory_indices, var_indices):
     """Auxiliary function to get the partial derivative of <A^2>."""
     pdA2 = 0
 
-    if non_involutory:
+    if non_involutory_indices:
         # compute the second derivative of non-involutory observables
         pdA2 = pdA2_fn(results[tape_boundary:])
 
         # For involutory observables (A^2 = I) we have d<A^2>/dp = 0.
-        involutory = set(var_idx) - set(non_involutory)
+        involutory = set(var_indices) - set(non_involutory_indices)
 
         if involutory:
-            pdA2 = _process_pda2_involutory(tape, pdA2, var_idx, non_involutory)
+            pdA2 = _put_zeros_in_pdA2_involutory(tape, pdA2, involutory)
+
     return pdA2
 
 
 def _create_variance_proc_fn(
-    tape, var_mask, var_idx, pdA_fn, pdA2_fn, tape_boundary, non_involutory
+    tape, var_mask, var_indices, pdA_fn, pdA2_fn, tape_boundary, non_involutory_indices
 ):
     """Auxiliary function to define the processing function for computing the
     derivative of variances using the parameter-shift rule.
 
     Args:
         var_mask (list): The mask of variance measurements in the measurement queue.
-        var_idx (list): The indices of variance measurements in the measurement queue.
+        var_indices (list): The indices of variance measurements in the measurement queue.
         pdA_fn (callable): The function required to evaluate the analytic derivative of <A>.
         pdA2_fn (callable): If not None, non-involutory observables are
             present; the partial derivative of <A^2> may be non-zero. Here, we
             calculate the analytic derivatives of the <A^2> observables.
         tape_boundary (callable): the number of first derivative tapes used to
             determine the number of results to post-process later
-        non_involutory (list): the indices in the measurement queue of all non-involutory
+        non_involutory_indices (list): the indices in the measurement queue of all non-involutory
             observables
     """
 
     def processing_fn(results):
         # We need to expand the dimensions of the variance mask,
         # and convert it to be the same type as the results.
-        res = results[0]

```

Comment:
On another look, this section served no direct purpose with our results being tuples/single array objects.